src/mlpack/methods/reinforcement_learning/worker/one_step_sarsa_worker.hpp

```diff
@@ -0,0 +1,250 @@
+  { reset(); }
+
+  /**
+   * Initialize the worker.
+   * @param learningNetwork The shared network.
+   */
+  void Initialize(NetworkType& learningNetwork)
+  {
+    updater.Initialize(learningNetwork.Parameters().n_rows,
+        learningNetwork.Parameters().n_cols);
+    // Build local network.
+    network = learningNetwork;
+  }
+
+  /**
+   * The agent will execute one step.
+   *
+   * @param learningNetwork The shared learning network.
+   * @param targetNetwork The shared target network.
+   * @param totalSteps The shared counter for total steps.
+   * @param policy The shared behavior policy.
+   * @param totalReward This will be the episode return if the episode ends
+   *     after this step. Otherwise this is invalid.
+   * @return Indicate whether current episode ends after this step.
+   */
+  bool Step(NetworkType& learningNetwork,
+            NetworkType& targetNetwork,
+            size_t& totalSteps,
+            PolicyType& policy,
+            double& totalReward)
+  {
+    // Interact with the environment.
+    if (action == ActionType::size)
+    {
+      // Invalid action means we are at the beginning of an episode.
+      arma::colvec actionValue;
+      network.Predict(state.Encode(), actionValue);
+      action = policy.Sample(actionValue, deterministic);
+    }
+    StateType nextState;
+    double reward = environment.Sample(state, action, nextState);
+    bool terminal = environment.IsTerminal(nextState);
+    arma::colvec actionValue;
+    network.Predict(nextState.Encode(), actionValue);
+    ActionType nextAction = policy.Sample(actionValue, deterministic);
+
+    episodeReturn += reward;
+    steps++;
+
+    terminal = terminal || steps >= config.StepLimit();
+    if (deterministic)
+    {
+      if (terminal)
+      {
+        totalReward = episodeReturn;
+        reset();
+        // Sync with latest learning network.
+        network = learningNetwork;
+        return true;
+      }
+      state = nextState;
+      action = nextAction;
+      return false;
+    }
+
+    #pragma omp atomic
+    totalSteps++;
+
+    pending[pendingIndex++] =
+        std::make_tuple(state, action, reward, nextState, nextAction);
+
+    if (terminal || pendingIndex >= config.UpdateInterval())
+    {
+      // Initialize the gradient storage.
+      arma::mat totalGradients(learningNetwork.Parameters().n_rows,
+          learningNetwork.Parameters().n_cols);
+      for (size_t i = 0; i < pending.size(); ++i)
+      {
+        TransitionType &transition = pending[i];
+
+        // Compute the target state-action value.
+        arma::colvec actionValue;
+        #pragma omp critical
+        {
+          targetNetwork.Predict(
+              std::get<3>(transition).Encode(), actionValue);
+        };
+        double targetActionValue = actionValue[std::get<4>(transition)];
+        if (terminal && i == pending.size() - 1)
+          targetActionValue = 0;
+        targetActionValue = std::get<2>(transition) +
```

Comment:
Should we use an else case here? That way we can slightly simplify the expression if ``(terminal && i == pending.size() - 1)`` is true.